stylegan_code_finder/lightning_modules/doc_ufcn_lightning.py

```diff
@@ -30,16 +36,16 @@ def training_step(self, batch, batch_idx):
         return loss
 
     def validation_step(self, batch, batch_idx):
-        # this is the validation loop
+        super(DocUFCNSegmenter, self).validation_step(batch, batch_idx)
         segmentation_prediction = self.segmentation_network(batch['images'])
         batch_size, num_classes, height, width = segmentation_prediction.shape
         segmentation_prediction = segmentation_prediction.permute(0, 2, 3, 1)
-        segmentation_prediction = torch.reshape(segmentation_prediction, (batch_size * height * width, num_classes))
+        segmentation_prediction = torch.reshape(segmentation_prediction, (-1, num_classes))
 
         label_image = batch['segmented']
         label_image = label_image.permute(0, 2, 3, 1)
         label_image = label_image.reshape((-1,))
 
         loss = self.ce_loss(segmentation_prediction, label_image)
-        self.log('val_loss', loss)
+        self.log('val_loss', loss, sync_dist=True)

```

Comment:
I'm not sure if `sync_dist` is really needed here (however the documentation is also not super specific and only statest that it might cause overhead). If you're not 100% sure that you need it, maybe remove it.  
However, may you can look into the `on_epoch` flag and accumulate losses.